src/commonMain/kotlin/fr/acinq/lightning/channel/Closing.kt

```diff
@@ -83,15 +75,26 @@ data class Closing(
                                     //
                                     // Force-closing is our only option here, if we are in this state the channel was closing and it is too late
                                     // to negotiate a mutual close.
+                                    // The best funding tx candidate has been confirmed, we can forget alternative commitments.
                                     logger.info { "channel was confirmed at blockHeight=${watch.blockHeight} txIndex=${watch.txIndex} with a previous funding txid=${watch.tx.txid}" }
-                                    val (nextState, actions1) = this@Closing.copy(commitments = commitments1).run { spendLocalCurrent() }
+                                    val commitments2 = commitments1.copy(
```

Comment:
I'm not sure about that, especially removing the `inactive` commitments, I believe we should only remove the alternative active commitments with the same `fundingTxIndex`. We're not yet handling splice-RBF so it's fine, but it's probably worth noting that splice-RBF will need to look into this carefully.

Reply:
I think it works in all cases, even with RBF, because we only do this if one of the _latest_ funding tx has confirmed. This implies that all other funding txs (parents and rbf), are double spent.

Reply:
Good point, you're right! Since we're only doing that for the latest `fundingTxIndex`, there shouldn't be any risk:

- there are no commitments with a strictly higher `fundingTxIndex`
- all other commitments are either spent by this confirmed tx (if they have a strictly smaller `fundingTxIndex`) or double-spent by it (if they have the same `fundingTxIndex`), so none of them can impact our force-close